### ISSUE TITLE
[Rule change] prefer-match-file-name: ignore characters after the first dot

### ISSUE THREAD

**Issue by furaiev** [dart-code-checker/dart-code-metrics#435]
**What rule do you want to change?** prefer-match-file-name

**How will the change be implemented? (New option, new default behavior, etc.)?**
prefer-match-file-name rule will ignore characters after the first dot in the file name

**Please provide some example code that this change will affect:**

```dart
// file name:  some_widget.codegen.dart

class SomeWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    //...
  }
}
```

**What does the rule currently do for this code?** 
consider the filename as incorrect

**What will the rule do after it's changed?**
consider the filename as correct

**Comment by incendial** [comment]
Available in 4.2.1 🚀
